models/base_GAN.py

```diff
@@ -220,7 +220,23 @@ def optimizeParameters(self, input_batch, inputLabels=None):
                                         backward=True)
 
         lossD.backward(retain_graph=True)
-        finiteCheck(self.netD.module.parameters())
+
+        if isinstance(self.netD, torch.nn.DataParallel):
+            netD_params = self.netD.module.parameters()
+        else:
+            netD_params = self.netD.parameters()
```

Comment:
There is a function BaseGAN::getOriginalD() which points netD.module or netD depending on the nature of netD